### ISSUE TITLE
Discrepency between create/update $is_new logic between pull and push

### ISSUE THREAD

**Issue by charmoney** [MinnPost/object-sync-for-salesforce#107]
There is a discrepancy in how $is_new is handled between the pull and push classes. I'd like to recommend having both use the technique from push but would settle for consistency.

**push**

During a push, a SF Object will always be created when needed.

````
if ( true === $is_new ) {
...
} else {
...
} // End if(). this is the end of the if is_new stuff
````

**pull**

During a pull, a WP Object can only be created when when the SF CreatedDate is more recent than the last sync. Additionally, a WP Object can only be updated when the SF CreatedDate is older than the last sync. If neither is the case, the SF object will not sync and also not provide an error message.

````
// if this record is new as of the last sync, use the create trigger
if ( isset( $result['CreatedDate'] ) && $result['CreatedDate'] > $last_sync ) {
	$trigger = $this->mappings->sync_sf_create;
} else {
	$trigger = $this->mappings->sync_sf_update;
}
...
if ( true === $is_new && ( $sf_sync_trigger == $this->mappings->sync_sf_create ) ) { // trigger is a bit operator
...
} elseif ( false === $is_new && ( $sf_sync_trigger == $this->mappings->sync_sf_update ) ) { // the trigger is a bit operator; currently it will fail if strict
...
	// end of the if statement for is_new & update or create trigger
	// this keeps stuff from running too many times, and also keeps it from using the wrong methods. we don't need a generic } else { here at this time.
} // End if().
````

The pull logic is causing occasional significant issues. For one real world example, if an initial WP creation fails for any reason, it's impossible through normal plugin usage to ever create the new record.

**Comment by jonathanstegall** [comment]
@charmoney thanks for this. I'll try to look into this and test at least your real world example. I think I get the issue you're indicating, but probably need to see it as well.

I'd also be happy to review any pull requests. I don't remember off the top of my head why it is set up the way it is, but am happy to consider other methods.

**Comment by charmoney** [comment]
The other real world example we're running into is the following scenario:

1. Tie into the `object_sync_for_salesforce_pull_object_allowed` to only pull SF records meeting specific criteria. (In our case, SF Accounts with an active subscription syncing to WP Users.)
2. Disallow the object pull when it's created in SF. (E.g. subscription not yet active.) The WP object will correctly not be created.
3. Modify the SF Object to fulfill pull criteria. (E.g. subscription is now active.)
4. Wait for the pull cron to execute.
5. The SF Object is correctly allowed through the filter. But because the SF action was an update rather than create, the WP object will incorrectly not be created.

I'll see about putting together a pull request to separate the mapping settings from the object creation.

**Comment by jonathanstegall** [comment]
Fixed by @charmoney in #108
